src/coreclr/tools/Common/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -200,7 +200,26 @@ private EcmaModule AddModule(string filePath, string expectedSimpleName, bool us
                     throw new NotSupportedException($"Error: C++/CLI is not supported: '{filePath}'");
 #endif
 
-                    pdbReader = PortablePdbSymbolReader.TryOpenEmbedded(peReader, GetMetadataStringDecoder()) ?? OpenAssociatedSymbolFile(filePath, peReader);
+                    try
+                    {
+                        pdbReader = PortablePdbSymbolReader.TryOpenEmbedded(peReader, GetMetadataStringDecoder())
+                                    ?? OpenAssociatedSymbolFile(filePath, peReader);
+                    }
+                    catch (BadImageFormatException e)
+                    {
+                        if (e.Message.Equals("Invalid directory size."))
```

Comment:
There are many other ways that the debug symbols can be corrupted, we just have not seen them. Should we catch all exceptions here and print that the debug symbols are corrupted, and include the error message from the exception?

Reply:
> There are many other ways that the debug symbols can be corrupted, we just have not seen them. Should we catch all exceptions here and print that the debug symbols are corrupted, and include the error message from the exception?

None of the compiler is hardened for obfuscators. I wouldn't treat debug information very differently.

Reply:
(We might successfully open the debug information but there can be exceptions once we start looking into it.)